main.tf

```
@@ -90,19 +91,19 @@ resource "aws_lb_target_group" "default" {
 }
 
 resource "aws_lb_listener" "default" {
-  count             = var.tcp_enabled ? 1 : (var.udp_enabled ? 1 : 0)
+  count             = local.create_target_group  == true && var.tcp_enabled ? 1 : (local.create_target_group  == true && var.udp_enabled ? 1 : 0)
```

perhaps you meant
```suggestion
  count             = local.target_group_enabled  && var.tcp_enabled ? 1 : (local.create_target_group  == true && var.udp_enabled ? 1 : 0)
```

as well as for the rest of these changes